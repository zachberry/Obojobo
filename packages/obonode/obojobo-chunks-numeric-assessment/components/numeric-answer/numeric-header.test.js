import React from 'react'
import renderer from 'react-test-renderer'

import NumericHeader from './numeric-header'

describe('NumericHeader', () => {
	test('NumericHeader renders with requirement of `exact`', () => {
		const component = renderer.create(<NumericHeader requirement="exact" />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericHeader renders with requirement of `range`', () => {
		const component = renderer.create(<NumericHeader requirement="range" />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericHeader renders with requirement of `margin`', () => {
		const component = renderer.create(<NumericHeader requirement="margin" />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericHeader renders with requirement of `precise`', () => {
		const component = renderer.create(<NumericHeader requirement="precise" />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericHeader renders with default requirement', () => {
		const component = renderer.create(<NumericHeader requirement="" />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
