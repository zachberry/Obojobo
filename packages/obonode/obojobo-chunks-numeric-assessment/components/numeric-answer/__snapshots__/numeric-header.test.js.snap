// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericHeader NumericHeader renders with default requirement 1`] = `
<tr>
  <th>
    Requirement
  </th>
  <th>
    Answer
  </th>
</tr>
`;

exports[`NumericHeader NumericHeader renders with requirement of \`exact\` 1`] = `
<tr>
  <th>
    Requirement
  </th>
  <th>
    Answer
  </th>
</tr>
`;

exports[`NumericHeader NumericHeader renders with requirement of \`margin\` 1`] = `
<tr>
  <th>
    Requirement
  </th>
  <th>
    Type
  </th>
  <th>
    Answer
  </th>
  <th>
    Margin
  </th>
</tr>
`;

exports[`NumericHeader NumericHeader renders with requirement of \`precise\` 1`] = `
<tr>
  <th>
    Requirement
  </th>
  <th>
    Type
  </th>
  <th>
    Answer
  </th>
  <th>
    Precision
  </th>
</tr>
`;

exports[`NumericHeader NumericHeader renders with requirement of \`range\` 1`] = `
<tr>
  <th>
    Requirement
  </th>
  <th>
    Start
  </th>
  <th>
    End
  </th>
</tr>
`;
